### ISSUE TITLE
[BUG] Proxy Issue With Firefox

### ISSUE THREAD

**Issue by nowayX** [microsoft/playwright#4082]
**Context:**
- Playwright Version: 1.4.2
- Operating System: Windows10
- Node.js version: v12.19.0
- Browser: Firefox
- Extra: Nothing

  System:
    OS: Windows 10 10.0.19041
    Memory: 5.81 GB / 11.94 GB
  Binaries:
    Node: 12.19.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.8 - C:\Program Files\nodejs\npm.CMD

```
const { firefox } = require('playwright');

(async () => {

  const browser = await firefox.launch({
    headless: false,
    proxy: {
      server: 'http://mybackbone.io:80',
      username: user',
      password: 'pass'
    }


  });
  const page = await browser.newPage();
  await page.goto('http://ifconfig.me');
```

**
i have this error : 

(node:2624) UnhandledPromiseRejectionWarning: browserType.launch: Protocol error (Browser.setBrowserProxy): ERROR: failed to call method 'Browser.setBrowserProxy' with parameters {
  "type": "http",
  "bypass": [],
  "host": "p.webshare.io",
  "port": null,
  "username": "ukpclbnm-1",
  "password": "bye7r9emc5kc"
}
Expected "<root>.port" to be |number|; found |object| `null` instead. _dispatch@chrome://juggler/content/protocol/Dispatcher.js:67:15

And if i switch to Chromium everything are ok, No errors

**

**Comment by mxschmitt** [comment]
Could you try the latest development version with `npm install playwright@next`. It seems like already fixed with #3850.

**Comment by nowayX** [comment]
Ok it works ! Thank you
